lib/internal/process/promises.js

```diff
@@ -55,19 +70,7 @@ function setupPromises(scheduleMicrotasks) {
         hasBeenNotifiedProperty.set(promise, true);
         const uid = promiseToGuidProperty.get(promise);
         if (!process.emit('unhandledRejection', reason, promise)) {
-          const warning = new Error('Unhandled promise rejection ' +
-                                    `(rejection id: ${uid}): ${reason}`);
-          warning.name = 'UnhandledPromiseRejectionWarning';
-          warning.id = uid;
-          process.emitWarning(warning);
-          if (!deprecationWarned) {
-            deprecationWarned = true;
-            process.emitWarning(
-              'Unhandled promise rejections are deprecated. In the future, ' +
-              'promise rejections that are not handled will terminate the ' +
-              'Node.js process with a non-zero exit code.',
-              'DeprecationWarning');
-          }
+          emitWarning(uid, reason);
```

Comment:
To allow for more inlining. `emitPendingUnhandledRejections()` is used by `setImmediate()`.